bigquery/indexer.py

```diff
@@ -124,8 +124,8 @@ def main():
 
   f = open(os.path.join(args.config_dir, 'facet_fields.csv'))
   # Remove comments using jsmin.
-  csvString = jsmin.jsmin(f.read())
-  rows = csv.DictReader(iter(csvString.splitlines()), skipinitialspace=True)
+  csv_str = jsmin.jsmin(f.read())
+  rows = csv.DictReader(iter(csv_str.splitlines()), skipinitialspace=True)
```

Comment:
Could the indexer be made case-insensitive with regards to field names?
BigQuery is case-insensitive with regards to column names:

https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#case-sensitivity

I found that I needed to set the field_name value to match the case in the BigQuery schema. Otherwise I'd get:

```
Traceback (most recent call last):
  File "indexer.py", line 136, in <module>
    main()
  File "indexer.py", line 132, in main
    row['dataset_id'], row['table_name'], row['field_name'], row['readable_field_name'])
  File "indexer.py", line 109, in index_facet_field
    bulk(es, k)
  File "/usr/local/google/home/mbookman/virtualenv/indexer-bigquery/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 257, in bulk
    for ok, item in streaming_bulk(client, actions, **kwargs):
  File "/usr/local/google/home/mbookman/virtualenv/indexer-bigquery/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 180, in streaming_bulk
    client.transport.serializer):
  File "/usr/local/google/home/mbookman/virtualenv/indexer-bigquery/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 58, in _chunk_actions
    for action, data in actions:
  File "indexer.py", line 107, in <genexpr>
    } for _, row in df.iterrows())
  File "/usr/local/google/home/mbookman/virtualenv/indexer-bigquery/local/lib/python2.7/site-packages/pandas/core/series.py", line 623, in __getitem__
    result = self.index.get_value(self, key)
  File "/usr/local/google/home/mbookman/virtualenv/indexer-bigquery/local/lib/python2.7/site-packages/pandas/core/indexes/base.py", line 2574, in get_value
    raise e1
KeyError: 'age'

```

Reply:
Let's see how often this problem comes up. I assumed most people would use the right case (and not know that BQ is case-insensitive for columns).

Reply:
@melissachang I wouldn't assume that.  From the BQ googlegenomics examples that folks have used, some are lowercase as shown here:

https://github.com/googlegenomics/bigquery-examples/blob/master/1000genomes_phase3/sql/qc-metrics.sql

Reply:
@pgrosu Which columns are you referring to? For example, I see there are `call` and `alternate_bases` columns in that code, and those columns are also lowercase in BQ.

Reply:
@melissachang Sorry assumed differences between SQL field capitalization between the links to show equivalence by implication, indicating BQ schema case-insensitivity for columns.  Here's a better one:

https://github.com/googlegenomics/bigquery-examples/blob/6e7ab1918aec89aae682e69b6beb32a2d99a8881/pgp/sql/cgi_variants/allelic-frequency-comparison.sql

Notice that  `AF` and `af` are both referenced:

![bq-lowercase-uppercase-invariant](https://user-images.githubusercontent.com/6555937/37854337-e9c60ea4-2ec0-11e8-8dc8-0de358b92562.png)

~p

Reply:
Got it, thanks for the examples